modules/mobile/app/controllers/mobile/v1/immunizations_controller.rb

```diff
@@ -16,6 +16,13 @@ def index
 
       private
 
+      def sort_map(vaccine_map)
+        vaccine_map.sort_by do |immunization|
+          [[immunization[:date] ? 1 : 0, immunization[:date] ? -immunization[:date].to_i : nil],
```

Comment:
yes the 0/1 is just for where to place the nils. I also included your suggestion of the comments. I kept the sorting here in the V1 because that is where pagination is set up and FE is taking the data through here

Reply:
I'm not sure I understand what you mean about keeping it here. This v1 endpoint was created to support pagination, but the v0 endpoint is still in use by the FE for older clients that we still support. My point was that by only adding it here, the bug still exists for older clients.